osutil/syncdir.go

```diff
@@ -96,10 +108,14 @@ func EnsureDirState(dir, glob string, content map[string]*FileState) (changed, r
 		changed = append(changed, baseName)
 	}
 	// Delete phase (remove files matching the glob that are not in content)
-	matches, err := filepath.Glob(filepath.Join(dir, glob))
-	if err != nil {
-		sort.Strings(changed)
-		return changed, nil, err
+	var matches []string
+	for _, glob := range globs {
```

Comment:
That's being done several times. If we move this logic out into a function the original code pretty much doesn't change:
```
func globAny(path, globs []string) (ok bool, error err)
```